src/proto/single/resp.rs

```diff
@@ -118,6 +120,92 @@ pub async fn read_ok<R: AsyncBufRead + Unpin>(r: R) -> Result<(), Error> {
     Err(bad("server ok", &rr).into())
 }
 
+// ----------------------------------------------
+
+/// Faktory service stats.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[non_exhaustive]
+pub struct DataSnapshot {
+    /// Total number of job failures.
+    pub total_failures: u64,
+
+    /// Total number of processed jobs.
+    pub total_processed: u64,
+
+    /// Total number of enqueued jobs.
+    pub total_enqueued: u64,
+
+    /// Total number of queues.
+    pub total_queues: u64,
+
+    /// Queues stats.
+    ///
+    /// A mapping between a queue name and its size (number of jobs on the queue).
+    /// The keys of this map effectively make up a list of queues that are currently
+    /// registered in the Faktory service.
+    pub queues: HashMap<String, u64>,
```

Comment:
Do you think iterating over these in order would ever be interesting to callers? As in, is there an argument for making this a `BTreeMap` instead?

Reply:
I feel like the `Client::current_info` is more for monitoring tools, as so I actually can imagine a situation where they are fetching this data on the server within the framework supporting SSR (not necessarily this, but for the example's sake) and iterating over these in a, say, `view!` macro. Normally, upon submitting a task via UI and having been redirected to the list view page, people will expect the queues to preserve the same order. We can of course apply some sorting onto the `HashMap` entries at an earlier step , but we get it (at a low price) right away with `BTreeMap`. Not sure how contrived this example is though.

Another example would be  an alert tool (maybe a headless one), that iterates over the `DataSnapshot::queues` and creates an event in the number of jobs on an queue is more than the predefined alert threshold. In this case, though, the order is of no particular interest.

Wdyt?

Reply:
So, the main challenge is that because this field is `pub`, we can't change its type later. As such, I'd be inclined to make it a "maximally useful" type, even if it comes with a slight runtime overhead. So let's switch to `BTreeMap` :+1: